lib/HLSL/HLOperationLower.cpp

```diff
@@ -5670,7 +5670,20 @@ Value *TranslateAllocateRayQuery(CallInst *CI, IntrinsicOp IOP,
                                  HLObjectOperationLowerHelper *pObjHelper,
                                  bool &Translated) {
   hlsl::OP *hlslOP = &helper.hlslOP;
-  Value *refArgs[] = {nullptr, CI->getOperand(1)};
+  // upgrade to allocateRayQuery2 if there is a non-zero 2nd template arg
+  DXASSERT(CI->getNumArgOperands() == 3,
+           "hlopcode for allocaterayquery always expects 3 arguments");
+
+  llvm::Value *Arg = CI->getOperand(2);
```

Comment:
You would use `HLOperandIndex::kAllocateRayQueryRayQueryFlagsIdx` here.  Also, I think you want `getArgOperand`, don't you?
```suggestion
  llvm::Value *Arg =
      CI->getArgOperand(HLOperandIndex::kAllocateRayQueryRayQueryFlagsIdx);
```

Reply:
I guess they return the same operand, but we generally use `getArgOperand` when getting call args in this code.